plugin.audio.ampache/resources/lib/servers_manager.py

```diff
@@ -0,0 +1,198 @@
+    i_curr = serversDialog(serverData,ut.tString(30169))
+    if i_curr == False:
+        return
+    xbmc.executebuiltin("PlayerControl(Stop)")
+    serverData["current_server"] = i_curr
+    jsStorServer.save(serverData)
+    #clean cache_art, the server is different, so the cache is invalid
+    clean_cache_art()
+    #if we switch, reconnect
+    try:
+        ampacheConnect = ampache_connect.AmpacheConnect()
+        ampacheConnect.AMPACHECONNECT(showok=True)
+    except:
+        pass
+
+def addServer():
+    xbmc.log("AmpachePlugin::addServer" , xbmc.LOGDEBUG )
+    jsStorServer = json_storage.JsonStorage("servers.json")
+    serverData = jsStorServer.getData()
+    stnum = len(list(serverData["servers"]))
+    username = ""
+    password = ""
+    apikey = ""
+    enablepassword = True
+    is_api_key = False
+    tempd = {}
+    tempd[stnum] = {}
+    serverData["servers"].update(tempd)
+    servername = gui.getFilterFromUser(ut.tString(30170))
+    if servername == False:
+        return False
+    url = gui.getFilterFromUser(ut.tString(30171))
+    if url == False:
+        return False
+    dialog = xbmcgui.Dialog()
+    is_api_key = dialog.yesno(ut.tString(30189),ut.tString(30173))
+    if is_api_key == True:
+        apikey = gui.getFilterFromUser(ut.tString(30174))
+        if apikey == False:
+            return False
+    else:
+        username = gui.getFilterFromUser(ut.tString(30175))
+        if username == False:
+            return False
+        enablepassword = dialog.yesno(ut.tString(30189),ut.tString(30177))
+        if enablepassword == True:
+            password = gui.getFilterFromUser(ut.tString(30178))
+            if password == False:
+                return False
+    serverData["servers"][stnum]["name"] = servername
+    serverData["servers"][stnum]["url"] = url
+    serverData["servers"][stnum]["use_api_key"] = ut.int_to_strBool(is_api_key)
+    serverData["servers"][stnum]["username"] = username
+    serverData["servers"][stnum]["enable_password"] = ut.int_to_strBool(enablepassword)
+    serverData["servers"][stnum]["password"] = password
+    serverData["servers"][stnum]["api_key"] = apikey
+    jsStorServer.save(serverData)
+    showServerData(serverData["servers"][stnum])
+    return True
+    
+def deleteServer():
+    jsStorServer = json_storage.JsonStorage("servers.json")
+    serverData = jsStorServer.getData()
+    i_rem = serversDialog(serverData,ut.tString(30179))
+    if i_rem == False:
+        return False
+    dialog = xbmcgui.Dialog()
+    confirm = dialog.yesno(ut.tString(30189),ut.tString(30188))
+    if confirm:
+        del serverData["servers"][i_rem]
+        jsStorServer.save(serverData)
+        return True
+    else:
+        return False
+
+def modifyServer():
+    jsStorServer = json_storage.JsonStorage("servers.json")
+    serverData = jsStorServer.getData()
+    i = serversDialog(serverData,ut.tString(30180))
+    if i == False:
+        return
+    while True:
```

Comment:
This is blocking isnt it? If you exit Kodi while in this loop Kodi will hang on close.
Can you use while not xbmc.Monitor().abortRequested(): instead?

Reply:
Added this check in while True loop

>        if xbmc.Monitor().abortRequested():
 >           return

i think i will be ok